.github/workflows/_build.yml

```diff
@@ -0,0 +1,272 @@
+# **what?**
+# Build release artifacts and store them to S3 bucket if they do not already exist.
+#
+# Build artifacts get stored in S3 to a bucket with the following directory structure:
+#       "s3://<s3_bucket>/<org>/<repo>/<version>/<commit>/"
+#   ex: "s3://core-team-artifacts/dbt-labs/dbt-core/1.2.3/ce98e6f067d9fa63a9b213bf99ebaf0c29d2b7eb/"
+#
+# Inputs:
+#  sha: the commit to attache to this release
+#  version_number: version number for the release (ex: 1.2.3rc2)
+#  build_script_path: Path to the build script
+#  s3_bucket: Bucket name
+#  package_test_command: Command to use to check package runs
+#   
+# **why?**
+# Reusable and consistent build process.
+#
+# **when?**
+# Call when you have a commit to build and want to trigger a release.
+#
+
+name: Build
+
+on:
+  workflow_call:
+    inputs:
+      sha:
+        required: true
+        type: string
+      version_number:
+        required: true
+        type: string
+      build_script_path:
+        required: true
+        default: "scripts/build-dist.sh"
+        type: string
+      s3_bucket_name:
+        required: true
+        default: "core-team-artifacts"
+        type: string
+      package_test_command:
+        required: true
+        default: "dbt --version"
+        type: string
+    # pass through secrets so every repo can have their own and won't depend on a name
+    secrets:
+      AWS_ACCESS_KEY_ID:
+        description: AWS Access Key ID
+        required: true
+      AWS_SECRET_ACCESS_KEY:
+        description: AWS Access Key
+        required: true
+
+permissions:
+  contents: write
+  # this will be needed if we go with OIDC for auth instead of manageing secrets in github for AWS
+  # id-token: write  # https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-cloud-providers#adding-permissions-settings
+
+env:
+  # include commit in path in case release commit gets updates on subsequent runs
+  s3_bucket: "s3://${{ inputs.s3_bucket_name }}/${{ github.repository }}/${{ inputs.version_number }}/${{ inputs.sha }}"
+  artifact_retention_days: 2
+
+jobs:
+  log-inputs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Print variables
+        run: |
+            echo The last commit sha in the release: ${{ inputs.sha }}
+            echo The release version number: ${{ inputs.version_number }}
+            echo The build_script_path: ${{ inputs.build_script_path }}
+            echo The s3_bucket_name: ${{ inputs.s3_bucket_name }}
+            echo The package_test_command: ${{ inputs.package_test_command }}
+
+  build-changelog-path:
+    # the changelog uses semver per https://semver.org/ so we need to build the path to the file to use it in the release notes
+    runs-on: ubuntu-latest
+    outputs:
+      changelog_path: ${{ steps.set_existence.outputs.exists }}
+
+    steps:
+
+      - name: Check out the repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.sha }}
```

Comment:
You can check the tag out here then instead

Reply:
I want to make sure both the tag and the commit match what we expect for the above reason.